### PR TITLE
Sync with ballerina 0.990.3

### PR DESCRIPTION
**Pull request by Biruntha**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -48,25 +50,26 @@ amazons3:AmazonS3Configuration amazonS3Config = {
...
+amazons3:Client amazonS3Client = new(amazonS3Config);
+
+public function main(string... args) {
 
     string bucketName = "testBallerina";
     var createBucketResponse = amazonS3Client->createBucket(bucketName);
     if (createBucketResponse is amazons3:Status) {
-        //If successful, returns the status value as true.
-        boolean status = string.create(createBucketResponse.success);
+        // If successful, prints the status value as true.
```
Comment (MaryamZi) [inline README.md]:
```suggestion
        // If successful, prints the status of the operation.
```

We can also remove the `status` variable and print inline.

Comment (MaryamZi) [reply README.md]: Same for Module.md

File: amazons3/amazons3_utils.bal [hunk 2]
```diff
@@ -21,6 +21,7 @@ import ballerina/time;
 import ballerina/http;
 import ballerina/crypto;
 import ballerina/system;
+import ballerina/encoding;
```
Comment (Shan1024) [inline amazons3/amazons3_utils.bal]: Better to sort the imports in alphabetical order.

File: amazons3/Module.md [hunk 3]
```diff
@@ -54,98 +54,100 @@ amazons3:AmazonS3Configuration amazonS3Config = {
...
+        io:println("Error: ", createBucketResponse);
     }
 
     io:println("-----------------Calling getBucketList() ------------------");
     var getBucketListResponse = amazonS3Client->getBucketList();
     if (getBucketListResponse is amazons3:Bucket[]) {
         io:println("Listing all buckets: ");
         foreach var bucket in getBucketListResponse {
-           io:println("Bucket Name: " + bucket.name);
+            io:println("Bucket Name: " + bucket.name);
```
Comment (Shan1024) [inline amazons3/Module.md]:
```suggestion
            io:println("Bucket Name: ", bucket.name);
```

File: amazons3/Module.md [hunk 4]
```diff
@@ -54,98 +54,100 @@ amazons3:AmazonS3Configuration amazonS3Config = {
...
-       io:println("Object content: " + content);
-   } else {
-       io:println(getObjectResponse);
-   }
+    io:println("-----------------Calling getObject() ------------------");
+    var getObjectResponse = amazonS3Client->getObject(bucketName, "test.txt");
+    if (getObjectResponse is amazons3:S3Object) {
+        io:println(getObjectResponse);
+        string content = getObjectResponse.content;
+        io:println("Object content: " + content);
```
Comment (Shan1024) [inline amazons3/Module.md]:
```suggestion
        io:println("Object content: ", content);
```

File: amazons3/Module.md [hunk 5]
```diff
@@ -54,98 +54,100 @@ amazons3:AmazonS3Configuration amazonS3Config = {
...
     io:println("-----------------Calling getAllObjects() ------------------");
     var getAllObjectsResponse = amazonS3Client->getAllObjects(bucketName);
     if (getAllObjectsResponse is amazons3:S3Object[]) {
         io:println("Listing all object: ");
         foreach var s3Object in getAllObjectsResponse {
-            io:println("---------------------------------");
-            io:println("Object Name: " + s3Object.objectName);
-            io:println("Object Size: " + s3Object.objectSize);
+        io:println("---------------------------------");
+        io:println("Object Name: " + s3Object.objectName);
```
Comment (Shan1024) [inline amazons3/Module.md]:
```suggestion
        io:println("Object Name: ", s3Object.objectName);
```

File: amazons3/Module.md [hunk 6]
```diff
@@ -54,98 +54,100 @@ amazons3:AmazonS3Configuration amazonS3Config = {
...
+        io:println("Object Size: " + s3Object.objectSize);
```
Comment (Shan1024) [inline amazons3/Module.md]:
```suggestion
        io:println("Object Size: ", s3Object.objectSize);
```

File: README.md [hunk 7]
```diff
@@ -48,25 +50,26 @@ amazons3:AmazonS3Configuration amazonS3Config = {
...
+amazons3:Client amazonS3Client = new(amazonS3Config);
+
+public function main(string... args) {
 
     string bucketName = "testBallerina";
     var createBucketResponse = amazonS3Client->createBucket(bucketName);
     if (createBucketResponse is amazons3:Status) {
-        //If successful, returns the status value as true.
-        boolean status = string.create(createBucketResponse.success);
+        // If successful, prints the status of the operation.
```
Comment (MaryamZi) [inline README.md]:
```suggestion
        // If successful, print the status of the operation.
```

File: amazons3/Module.md [hunk 8]
```diff
@@ -54,98 +54,100 @@ amazons3:AmazonS3Configuration amazonS3Config = {
...
-The `createBucket` function creates a bucket.   
-If the creation was successful, the response from the `createBucket` function is a `Status` object with the success value. If the creation was unsuccessful, the response is an `error`. 
+The `createBucket` remote function creates a bucket.
+If the creation was successful, the response from the `createBucket` function is a `Status` object with the success value. If the creation was unsuccessful, the response is an `error`.
 
 ```ballerina
 var createBucketResponse = amazonS3Client->createBucket(bucketName);
 if (createBucketResponse is amazons3:Status) {
-    //If successful, returns the status value as true.
+    // If successful, prints the status of the operation.
```
Comment (MaryamZi) [inline amazons3/Module.md]:
```suggestion
    // If successful, print the status of the operation.
```

File: amazons3/Module.md [hunk 9]
```diff
@@ -54,98 +54,100 @@ amazons3:AmazonS3Configuration amazonS3Config = {
...
+amazons3:Client amazonS3Client = new(amazonS3Config);
 
+public function main(string... args) {
     string bucketName = "testBallerina";
     io:println("-----------------Calling createBucket() ------------------");
+    var createBucketResponse = amazonS3Client->createBucket(bucketName);
     if (createBucketResponse is amazons3:Status) {
-        //If successful, returns the status value as true.
-        boolean status = string.create(createBucketResponse.success);
+        // If successful, prints the status value as true.
```
Comment (MaryamZi) [inline amazons3/Module.md]:
```suggestion
        // If successful, print the status of the operation.
```

**Pull request merged**